moped-editor/src/components/DataGridPro/LookupAutocompleteComponent.js

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { Autocomplete, TextField } from "@mui/material";
+import { useGridApiContext } from "@mui/x-data-grid-pro";
+import FullWidthPopper from "src/components/FullWidthPopper";
+import { filterOptions } from "src/utils/autocompleteHelpers";
+
+/**
+ * Component for dropdown select using a lookup table as options
+ * @param {Number} id - row id in Data Grid
+ * @param {string} value - Field value
+ * @param {string} field - name of Field
+ * @param {Boolean} hasFocus - does field have focus in table
+ * @param {String} name - name of lookup table relationship
+ * @param {Array|Objects} options - the lookup table data
+ * @param {Object} autocompleteProps - props passed to the MUI Autcomplete Component
+ * @param {Boolean} fullWidthPopper - should component use custom Popper component
+ *
+ * @returns {React component}
+ */
+const LookupAutocompleteComponent = ({
+  id,
+  value,
+  field,
+  hasFocus,
+  name,
+  options,
+  fullWidthPopper,
+  autocompleteProps,
+}) => {
+  const apiRef = useGridApiContext();
+  const ref = React.useRef(null);
+
+  React.useEffect(() => {
+    if (hasFocus) {
+      ref.current.focus();
+    }
+  }, [hasFocus]);
+
+  const handleChange = (event, newValue) => {
+    apiRef.current.setEditCellValue({
+      id,
+      field,
+      value: newValue,
+    });
+  };
+
+  const defaultGetOptionLabel = (option) => option[`${name}_name`];
+
+  const defaultIsOptionEqualToValue = (value, option) =>
```

Comment:
this function and the one above it look like utilities that could get declared/imported elsewhere 😇

Reply:
I started trying to declare them in the utils folder but then got tripped up with trying to pass `name` to the function since Autocomplete is expecting only `options` https://mui.com/material-ui/api/autocomplete/#autocomplete-prop-getOptionLabel 

Chatted with Mike, I am going to wrap them in a useCallback